sw/device/tests/sim_dv/data_integrity_escalation_reset_test.c

```diff
@@ -13,11 +16,54 @@
 // the interrupt, and it is also cleared after reset.
 //
 // As a backup the aon timer is programmed to bark and bite, but these are
-// expected not to happen since the escalation takes precedence.
+// expected not to happen since the escalation takes precedence. In particular,
+// the timeouts are set so that bark and bite happen during the escalation phase
+// where lc_escalate_en is asserted, which halts the watchdog timers.
 //
-// The alert_handler is programmed to escalate on the first interrupt, so the
-// regular ISR is not necessarily run and the load integrity error handler may
-// be beat by the NMI.
+// The alert_handler is programmed to escalate on the first alert event it
+// registers. It is also programmed to do nothing in its first escalation phase
+// so that the exception handlers and regular ISRs can execute before the NMI is
+// triggered in the subsequent escalation phase.
+//
+// In summary, the test comprises the following steps:
+//
+// 0) SV sequence waits until test is running and the test function has been
+// loaded to SRAM. It then randomly decides where to inject a fault, and
+// communicates this to the program via symbol overwrites.
+//
+// 1) Test sets up peripherals
+//
+// 2) Either read a corrupted memory location in main or retention SRAM, or
+// jump to a corrupted test function in SRAM.
+//
+// 3) Alert handler class A escalates right away when seeing the first alert
```

Comment:
We tend to use class A... could you use any other one just for kicks?

Reply:
Sure, why not - search replaced everything to class D.

Reply:
Will recommend picking the irq class randomly using `rand_testutils_gen32_range(0, 3)`.

Reply:
Good idea - I changed as suggested.